src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalAdjuster adjuster) {
+        return (InternationalFixedDate) adjuster.adjustInto(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalField field, final long newValue) {
+        if (field instanceof ChronoField) {
+            ChronoField f = (ChronoField) field;
+            getChronology().range(f).checkValidValue(newValue, f);
+
+            if (f == ChronoField.DAY_OF_MONTH || f == ChronoField.MONTH_OF_YEAR) {
+                if (newValue == 0) {
+                    return createYearDay(getProlepticYear());
+                }
+
+                if (newValue == -1) {
+                    return createLeapDay(getProlepticYear());
+                }
+            }
+
+            int dom = isYearDay() ? 21 : (getCalculatedDayOfMonth() / 7) * 7;
+            int d = day < 1 ? 1 : day % 7;
+            int nval = (int) newValue;
+
+            switch (f) {
+                case DAY_OF_WEEK:
+                case ALIGNED_DAY_OF_WEEK_IN_MONTH:
+                case ALIGNED_DAY_OF_WEEK_IN_YEAR:
+                    if (newValue == 0) {
+                        return this;
+                    }
+                    return resolvePreviousValid(getProlepticYear(), getMonth(), dom + nval);
+                case ALIGNED_WEEK_OF_MONTH:
+                    if (newValue == 0) {
+                        return this;
+                    }
+                    return resolvePreviousValid(getProlepticYear(), getMonth(), (nval - 1) * 7 + d);
+                case ALIGNED_WEEK_OF_YEAR:
+                    if (newValue == 0) {
+                        return this;
+                    }
+                    return ofYearDay(getProlepticYear(), (nval - 1) * 7 + d);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.with(field, newValue);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plus(final TemporalAmount amount) {
+        return (InternationalFixedDate) amount.addTo(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plus(final long amountToAdd, final TemporalUnit unit) {
+        if (unit instanceof ChronoUnit) {
+            ChronoUnit f = (ChronoUnit) unit;
+
+            switch (f) {
+                case WEEKS:
+                    return plusWeeks(amountToAdd);
+                case MONTHS:
+                    return plusMonths(amountToAdd);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.plus(amountToAdd, unit);
+    }
+
+
+    private InternationalFixedDate plusWeeks(final long weeks) {
```

Comment:
`plusWeeks(...)` will return incorrect results once 7 non-week days have elapsed, even though the calculated day-of-week will be the same. (actually, earlier, if you pass more than one non-week day).  Add the following test cases at the bottom of the plus data. 

    {2012, 6, 21, 52 + 1, WEEKS, 2013, 6, 28},
    {2013, 6, 21, 6 * 52 + 1, WEEKS, 2019, 6, 28},

 I recommend you use `prolepticWeek()`, similar to what I did in `DiscordianDate`.